spring-boot-modules/spring-boot-crud/src/test/java/com/baeldung/entityDtoDifferences/controller/UserControllerIntegrationTest.java

```diff
@@ -0,0 +1,81 @@
+package com.baeldung.entityDtoDifferences.controller;
+
+import com.baeldung.entityDtoDifferences.entity.Book;
+import com.baeldung.entityDtoDifferences.entity.User;
+import com.baeldung.entityDtoDifferences.repository.UserRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UserControllerIntegrationTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    public void beforeEach() {
+        userRepository.deleteAll();
+    }
+
+    @Test
+    public void givenUsersAreExisting_whenGetUsers_thenUsersAreReturned() throws Exception {
+        //given
+        Book book1 = new Book("Book1", "Author1");
+        Book book2 = new Book("Book2", "Author2");
+        User savedUser = userRepository.save(new User("John", "Doe", "123 Main St", Arrays.asList(book1, book2)));
+        String expectedJson = String.format(
+                "[{\"ID\":%s,\"FIRST_NAME\":\"John\",\"LAST_NAME\":\"Doe\",\"BOOKS\":[{\"NAME\":\"Book1\",\"AUTHOR\":\"Author1\"},{\"NAME\":\"Book2\",\"AUTHOR\":\"Author2\"}]}]",
+                savedUser.getId()
+        );
+        //when
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get("/users")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        //then
+        assertEquals(expectedJson, result.getResponse().getContentAsString());
+    }
+
+    @Test
+    public void givenValidUser_whenPostUser_thenUserIsCreated() throws Exception {
+        //given
+        String user = "{\"FIRST_NAME\":\"John\",\"LAST_NAME\":\"Doe\",\"BOOKS\":[{\"NAME\":\"Book1\",\"AUTHOR\":\"Author1\"},{\"NAME\":\"Book2\",\"AUTHOR\":\"Author2\"}]}";
+        //when
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/users")
+                        .content(user)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andReturn();
```

Comment:
This still does not look like our standard formatter, which you can find here:
https://github.com/Baeldung/dev-settings/tree/main/javaCodeStyle

Reply:
I'm using the formatter ... I've modified the indentation for this specific code block and formatted the code ... so this is the output of our standard formatter

Reply:
@ovidiumihaitacu which IDE are you using? Here is what I see when I apply the Baeldung formatter to this class:

<img width="986" alt="Screenshot 2023-11-30 at 09 36 47" src="https://github.com/eugenp/tutorials/assets/6877541/ec24ac29-342e-43a8-9994-1d7a08e6f4f4">

<img width="1376" alt="Screenshot 2023-11-30 at 09 38 42" src="https://github.com/eugenp/tutorials/assets/6877541/c02f8650-ce1f-4120-83bb-38841ea0e8b1">

Reply:
@theangrydev Same as you, Intellij. This is my config:
<img width="1470" alt="image" src="https://github.com/eugenp/tutorials/assets/138307181/9b6fbbc8-8ae7-45c4-b22d-f4508d82b313">

I reran the format and this is the outcome (now it's different than before, it formatted the code to have one-liners):
<img width="1470" alt="image" src="https://github.com/eugenp/tutorials/assets/138307181/01e113f5-700a-498e-a01d-f1725d67b232">

It's strange...